docker/Dockerfile.sys-mgmt-agent

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright (c) 2018
+# Dell
+# Cavium
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+FROM golang:1.11.2-alpine3.7 AS builder
+
+# Build the SMA bits by fetching the source from Github.
+WORKDIR /go/src/github.com/edgexfoundry/edgex-go
+
+# The main mirrors are giving us timeout issues on builds periodically.
+# So we can try these.
+RUN echo http://nl.alpinelinux.org/alpine/v3.7/main > /etc/apk/repositories; \
+    echo http://nl.alpinelinux.org/alpine/v3.7/community >> /etc/apk/repositories
+
+RUN apk update && apk add make glide git
+COPY . .
+
+RUN make prepare
+RUN make cmd/sys-mgmt-agent/sys-mgmt-agent
+
+# Build the golang "executor" bits (an SMA-related reference implementation) by fetching the source from Github.
+WORKDIR /go/src/github.com/docker-compose
+RUN git clone https://github.com/akramtexas/docker-compose
```

Comment:
I'm more just commenting here because I want to raise attention to this point on why I think doing it this way is a bad idea. Here, in order to include the docker-compose executable from this external repo, we have to clone another external git repo inside this docker build file, which obviously works right now but has the following issues:

* you will now have duplicated build code for both this repo here and the docker-compose repo as well (I assume that repo will be adopted into `edgexfoundry` and as such will need it's own build code in that repository separate from this repo. That code is going to be out-of-sync with this code because likely no one is going to independently test/run that repo, they will only ever use the version in edgex-go and that's the sort of thing that leads to bit-rotting and old confusing code that doesn't run anymore. Having all the build code for a single thing in one place is much better. All that being said, you could also just put the build code inside a Makefile in the other repo and call `make build` here and that would be better
* because you are cloning this repo here, and because I assume that this other repo will need to be released with the rest of EdgeX come release day, the other repo will have it's branch change and someone will need to remember to come in to the release branch on this repo and change this git clone to checkout the correct branch. Building the other code inside this repo removes that problem entirely.
* testing changes to the other, external repo will become burdensome and difficult for reviewers because they will need to change this line in the docker compose file to checkout your branch (or otherwise build up all of edgex-go in their own docker containers) in order to build a working container which can run the things needed. This may not seem like a big deal now, but we want to make it easier to test changes to components of EdgeX. For example, it means that writing automated CI solutions that can run integration tests that would run against a specific PR that is opened would need to perform this kind of modification step in order to build the container in this repo and test it which is highly unfortunate and difficult for the devops team to setup. 
* I also highly discourage pulling in code that we build like this from a github repo under your personal account. I think before this PR can be merged that code has to be pulled into edgexfoundry where it can be reviewed and will have a higher visibility than under your personal account. Additionally this bit brings up the point about who will review (or has reviewed) this external go code? If this PR was merged right now and my assumption that no one has reviewed your code there yet then we are essentially bringing in completely un-reviewed code and building it.